compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1743,28 +1747,28 @@ getMatmulCacheTileSizesForShape(ArrayRef<int64_t> inputTileSizes,
 /// Sets the lowering configuration for dispatch region with root op that
 /// implements the contraction operation interface.
 static LogicalResult
-setRootConfig(mlir::FunctionOpInterface entryPointFn,
-              linalg::ContractionOpInterface contractionOp) {
-  assert(!getLoweringConfig(contractionOp) &&
-         "expected lowering_config is not set");
-  auto linalgOp = cast<linalg::LinalgOp>(contractionOp.getOperation());
+setContractionRootConfig(mlir::FunctionOpInterface entryPointFn,
+                         linalg::LinalgOp linalgOp) {
+  assert(!getLoweringConfig(linalgOp) && "expected lowering_config is not set");
```

Comment:
I know you check before calling, but I think it would be robust towards future use to add an assert here that this linalg op implements the `ContractionOpInterface`. It's also mentioned in the docstring above that the op should implement the interface.

Reply:
I'm now a little concern about such assertion, because I've seen that the compile time is very bad in debug build. Checking if an op implements the interface or not is not very cheap, like checking if it has the trait or not. It looks at the body and infer contraction dimensions. If it is a constant time check, like `isa<>`, I'm happy to do so.